.travis.yml

```diff
@@ -43,20 +43,22 @@ matrix:
     - php: hhvm
     - php: nightly
 
-before_script:
+before_install:
   - (phpenv config-rm xdebug.ini || exit 0)
   - mkdir -p ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d && echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
   - composer selfupdate
   - composer config -q -g github-oauth.github.com $GITHUB_OAUTH_TOKEN
-  - composer global require phpunit/phpunit:@stable fabpot/php-cs-fixer --no-update
-  - composer global update --prefer-dist --no-interaction
+  - if [ "$TARGET" = "cs_dry_run" ]; then composer global require fabpot/php-cs-fixer --no-update; fi;
+  - if [ "$TARGET" = "test" ]; then composer global require phpunit/phpunit:@stable --no-update; fi;
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
-  - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
   - export PATH=$HOME/.local/bin:$PATH
-  - pip install -r Resources/doc/requirements.txt --user `whoami`
 
-script:
- - make $TARGET
+install:
+  - if [ "$TARGET" != "docs" ]; then composer global update --prefer-dist --no-interaction; fi;
```

Comment:
Please note that it made quite a difference:

 * **17min 11sec** - Test with these changes (https://travis-ci.org/sonata-project/SonataDoctrinePhpcrAdminBundle/builds/127749164)
 * **22min 24sec** - Test without these changes (https://travis-ci.org/sonata-project/SonataDoctrinePhpcrAdminBundle/builds/127751689)